homeassistant/components/matter/entity.py

```diff
@@ -114,23 +115,27 @@ def __init__(
         # prefer the label attribute for the entity name
         # Matter has a way for users and/or vendors to specify a name for an endpoint
         # which is always preferred over a standard HA (generated) name
-        for attr in (
-            clusters.FixedLabel.Attributes.LabelList,
-            clusters.UserLabel.Attributes.LabelList,
+        if (
+            not hasattr(entity_info.entity_description, "ignore_matter_label_name")
+            or entity_info.entity_description.ignore_matter_label_name is False
```

Comment:
This is really the main change - if the attribute ignore_matter_label_name is set to True, then don't use FixedLable or UserLable. For backward-compatibility, if the attribute isn't present, the code operates as it previously did.